src/sql/workbench/browser/parts/views/customView.ts

```diff
@@ -401,59 +529,74 @@ export class CustomTreeView extends Disposable implements ITreeView {
 
 	refresh(elements?: ITreeItem[]): Promise<void> {
 		if (this.dataProvider && this.tree) {
-			elements = elements || [this.root];
+			if (!elements) {
+				elements = [this.root];
+				// remove all waiting elements to refresh if root is asked to refresh
+				this.elementsToRefresh = [];
+			}
 			for (const element of elements) {
-				element.children = null; // reset children
+				element.children = undefined; // reset children
 			}
 			if (this.isVisible) {
 				return this.doRefresh(elements);
 			} else {
-				this.elementsToRefresh.push(...elements);
+				if (this.elementsToRefresh.length) {
+					const seen: Set<string> = new Set<string>();
+					this.elementsToRefresh.forEach(element => seen.add(element.handle));
+					for (const element of elements) {
+						if (!seen.has(element.handle)) {
+							this.elementsToRefresh.push(element);
+						}
+					}
+				} else {
+					this.elementsToRefresh.push(...elements);
+				}
 			}
 		}
-		return Promise.resolve(null);
+		return Promise.resolve(undefined);
 	}
```

Comment:
return Promise.resolve(); ? This is returning a Promise<void>

Reply:
Hm, I guess this is allowed and likely isn't an issue but still seems weird when return Promise.resolve() works fine. No huge deal though

---
In reply to: [305935181](https://github.com/microsoft/azuredatastudio/pull/6442#discussion_r305935181) [](ancestors = 305935181)